internal/exit/exit.go

```diff
@@ -33,10 +33,12 @@ func OnError(err error) {
 }
 
 // WithMessage will print the message and quit the program with an error code.
-func WithMessage(message string) {
-	fmt.Fprintln(os.Stderr, message)
-	printVersion()
-	os.Exit(1)
+func WithMessage(message string, err error) {
```

Comment:
I think we should leave this function as is - there are cases where we want to just print a message and exit in the absence of an error. For the case where we want to augment an error with an additional message, I suggest an `OnErrorWithMessage` function.